vpr/src/util/vpr_utils.cpp

```diff
@@ -2356,3 +2356,80 @@ void print_timing_stats(std::string name,
             current.num_full_hold_updates - past.num_full_hold_updates,
             current.num_full_setup_hold_updates - past.num_full_setup_hold_updates);
 }
+
+void alloc_legal_placement_locations(std::vector<std::vector<std::vector<t_pl_loc>>>& legal_pos,
```

Comment:
Please add a TODO comment: replace num_legal_pos by using the .size() of legal_pos instead. We'll leave this for the future but good to mark it down.